src/App.tsx

```diff
@@ -1,25 +1,131 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+function getNewTodoId(todos: Todo[]) {
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: getUser(todo.userId),
+  };
+});
+
+export const App: React.FC = () => {
+  const [initialTodo, setInitialTodo] = useState<Todo[]>(todos);
+
+  const [userId, setUserId] = useState(0);
+  const [selectErrorMessage, setSelectErrorMessage] = useState(false);
+
+  const [title, setTitle] = useState('');
+  const [titleErrorMessage, setTitleErrorMessage] = useState(false);
+
+  const addNewTodo = ({ id, ...data }: Todo) => {
+    const newTodo = {
+      id: getNewTodoId(initialTodo),
+      ...data,
+    };
+
+    setInitialTodo((currentTodo) => [...currentTodo, newTodo]);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleErrorMessage(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setSelectErrorMessage(false);
+  };
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+    setTitleErrorMessage(false);
+    setSelectErrorMessage(false);
+  };
+
+  const handleSubmitForm = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setTitleErrorMessage(!title);
+    setSelectErrorMessage(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    addNewTodo({
+      id: 0,
+      title,
+      userId,
+      completed: false,
+      user: getUser(userId),
+    });
+
+    resetForm();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmitForm}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            value={title}
+            placeholder="Enter a title"
+            data-cy="titleInput"
+            onChange={handleTitleChange}
+          />
+
+          {titleErrorMessage && (
+            <span className="error">
+              Please enter a title
```

Comment:
You can save the error text in state and use here instead of the hardcoded value